scripts/format-cpp.pl

```diff
@@ -91,63 +66,71 @@
         }
         close(TO_ASTYLE);
         waitpid($pid,0);
+        waitpid($pid_style, 0);
     }
     else{
         # child staf
         close(TO_ASTYLE);
 
-        if(!open(OUT,">$out")){
-            print "Can't open output file: $out\n";
-            exit -1;
-        }
+        my $formattedCode = '';
         my($line)='';
         while(<FROM_ASTYLE>){
             $line = $line.$_;
             if(output_filter(\$line)==0){
                 next;
             }
-            print OUT $line;
+            $formattedCode .= $line;
             $line = '';
         }
         if($line){
-            print OUT $line;
+            $formattedCode .= $line;
         }
-        close(OUT);
-        exit 0;
-    }
 
-    $out = shift @ARGV;
-}
+        my $originalCode = &slurpFile($out);
 
-# renames while ensuring the destination is not clobbered
-sub safeRename
-{
-    my ($from, $to) = @_;
-    die() if -e $to;
-    rename($from, $to) or die("Failed to rename $from to $to: $!, stopped");
-}
+        if (!length $formattedCode) {
+            warn("ERROR: Running astyle produced no output while formatting $out\n".
+                 "    astyle command: $ASTYLE_BIN\n");
+            print $originalCode;
```

Comment:
I doubt we should restore the original (and possibly unformatted) output. Instead, we should just report the error and exit.

The current PR code kind of preserves the official behavior that was probably motivated by the combination of buggy astyle versions and lack of content preservation guarantees. These days, modern astyle does not fail very often, and git checks guarantee that nothing is lost. Thus, we can and, hence, should treat such failures like most source-maintenance.sh failures -- report and exit, leaving any artifacts as they were.

If you agree _and_ think that this improvement/simplification belongs to _this_ PR, then please let me know, and I will implement it.

This applies to other `print $originalCode` places, of course.